gluten-data/src/main/scala/io/glutenproject/expression/GlutenExpressionTransformer.scala

```diff
@@ -54,3 +55,37 @@ class GlutenNamedStructTransformer(substraitExprName: String,
     ExpressionBuilder.makeScalarFunction(functionId, expressionNodes, typeNode)
   }
 }
+
+
+class GlutenHashExpressionTransformer(
```

Comment:
nit: better to make it a case class and remove `new` at the instantiating place.

Reply:
may must make it case class, otherwise the equal() function will return wrong result and then operator such as ReuseExchange cannot make sense